externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/RowSet.scala

```diff
@@ -0,0 +1,320 @@
+      } else {
+        ret.add(idx, row(ordinal).asInstanceOf[T])
+      }
+      idx += 1
+    }
+    ret
+  }
+
+  private def toTColumnValue(
+      ordinal: Int,
+      row: List[Any],
+      types: List[Column],
+      timeZone: ZoneId): TColumnValue = {
+
+    types(ordinal).getType match {
+      case BOOLEAN =>
+        val boolValue = new TBoolValue
+        if (row(ordinal) != null) boolValue.setValue(row(ordinal).asInstanceOf[Boolean])
+        TColumnValue.boolVal(boolValue)
+
+      case TINYINT =>
+        val byteValue = new TByteValue
+        if (row(ordinal) != null) byteValue.setValue(row(ordinal).asInstanceOf[Byte])
+        TColumnValue.byteVal(byteValue)
+
+      case SMALLINT =>
+        val tI16Value = new TI16Value
+        if (row(ordinal) != null) tI16Value.setValue(row(ordinal).asInstanceOf[Short])
+        TColumnValue.i16Val(tI16Value)
+
+      case INTEGER =>
+        val tI32Value = new TI32Value
+        if (row(ordinal) != null) tI32Value.setValue(row(ordinal).asInstanceOf[Int])
+        TColumnValue.i32Val(tI32Value)
+
+      case BIGINT =>
+        val tI64Value = new TI64Value
+        if (row(ordinal) != null) tI64Value.setValue(row(ordinal).asInstanceOf[Long])
+        TColumnValue.i64Val(tI64Value)
+
+      case REAL =>
+        val tDoubleValue = new TDoubleValue
+        if (row(ordinal) != null) tDoubleValue.setValue(row(ordinal).asInstanceOf[Float])
+        TColumnValue.doubleVal(tDoubleValue)
+
+      case DOUBLE =>
+        val tDoubleValue = new TDoubleValue
+        if (row(ordinal) != null) tDoubleValue.setValue(row(ordinal).asInstanceOf[Double])
+        TColumnValue.doubleVal(tDoubleValue)
+
+      case VARCHAR =>
+        val tStringValue = new TStringValue
+        if (row(ordinal) != null) tStringValue.setValue(row(ordinal).asInstanceOf[String])
+        TColumnValue.stringVal(tStringValue)
+
+      case _ =>
+        val tStrValue = new TStringValue
+        if (row(ordinal) != null) {
+          tStrValue.setValue(
+            toHiveString((row(ordinal), types(ordinal).getType), timeZone))
+        }
+        TColumnValue.stringVal(tStrValue)
+    }
+  }
+
+  /**
+   * A simpler impl of Trino's toHiveString
+   */
+  def toHiveString(dataWithType: (Any, String), timeZone: ZoneId): String = {
+    dataWithType match {
+      case (null, _) =>
+        // Only match nulls in nested type values
+        "null"
+
+      case (sd: java.sql.Date, DATE) =>
+        dateFormatter.format(sd.toLocalDate)
+
+      case (d: Date, DATE) =>
+        dateFormatter.format(d.toInstant)
+
+      case (ld: LocalDate, DATE) =>
+        dateFormatter.format(ld)
+
+      case (t: Timestamp, TIMESTAMP) =>
+        timestampFormatter.withZone(timeZone).format(t.toInstant)
+
+      case (i: Instant, TIMESTAMP) =>
+        timestampFormatter.withZone(timeZone).format(i)
+
+      case (t: Timestamp, TIMESTAMP_WITH_TIME_ZONE) =>
+        timestampWithZoneFormatter.withZone(timeZone).format(t.toInstant)
```

Comment:
`select timestamp '2012-11-11 ... ' at time zone 'UTC'`, in this case, we will reformat to `Asia/Shanghai`

Reply:
I had made a mistake. For `time` `timestamp`..., trino client always return as `(String, TIME)`. We no need to format it, because it's already formatted.

Reply:
![image](https://user-images.githubusercontent.com/17537134/145968080-346dd4c8-ac26-4eea-802f-7f6116f9ca09.png)